### PR TITLE
User Sign In and Logout 

### LINKED ISSUE #1: User can Login
As a visitor when I visit '/' I want to register so I can use the site. 

- [x] Create user table with name, email, and token columns
- [x] Implement google Omniauth strategy
- [x] Route for login page is /login
- [x] User should be redirected to /dashboard after successful login
- [x] Build a dashboard view

### PR DESCRIPTION
**Pull request by SMJ289**
## What functionality does this accomplish?
- Add routes for logging in and out
- Add User model and database migration; Added uniqueness validations to prevent duplicate accounts
- Add WelcomeController
- Add SessionsController
- Add welcome controller
- Add NavBar with dynamic Sign In and Logout link
- Add bootstrap, jquery, and Figaro gems
- Add Google Omni-auth setup

closes #1 #2 
**Description:**
Adds user sign in and logout functionality using Google OAuth.
### Rubocop:
- [x] No offenses
- [ ] Offenses (number of offenses: _ )
### Test Coverage Percentage: 100%
- [x] No tests have been changed
- [ ] Some tests have been changed
- [ ] All of the tests have been changed(Please describe what in the world happened):
## Checklist:
- [x] My code has no unused/commented out code
- [x] I have reviewed my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have fully tested my code
- [ ] I have partially tested my code (please explain why):
## Helpful Resources:
* Resource link AND small description of info gathered/learned

### CODE REVIEW

File: db/migrate/20200528214834_create_users.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class CreateUsers < ActiveRecord::Migration[5.1]
+  def change
+    create_table :users do |t|
+      t.string :name
```
Comment (javier-aguilar) [inline db/migrate/20200528214834_create_users.rb:4]: So are we currently putting the first and last name in the name column?

**Comment by javier-aguilar** [comment]
What were the results of rubocop? Otherwise, everything looks solid 👍

File: db/migrate/20200528214834_create_users.rb (hunk 1 above)
Comment (danielbldr) [reply db/migrate/20200528214834_create_users.rb:4]: Right now we just have the first name - there was a key for the last name, but I noticed my specific last name was blank. I didn't think there would be a use for the last name in our app, but we could definitely add a column if you think its best

**Comment by danielbldr** [comment]
> What were the results of rubocop? Otherwise, everything looks solid 👍

We have not configured rubocop yet

File: db/migrate/20200528214834_create_users.rb (hunk 1 above)
Comment (SMJ289) [reply db/migrate/20200528214834_create_users.rb:4]: We just have it getting the first name or 'given_name' attached to the Google account. Do you think we should get the last name or 'family_name' from Google as well?

**Comment by SMJ289** [comment]
Yeah, we need to set up Rubocop configuration but I'll add a checkbox to the PR template as a reminder.

File: db/migrate/20200528214834_create_users.rb (hunk 1 above)
Comment (javier-aguilar) [reply db/migrate/20200528214834_create_users.rb:4]: Ah. Makes sense. Our app shouldn't need the last name really

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class User < ApplicationRecord
+  validates :name, presence: true
+  validates :email, presence: true
+  validates :uid, presence: true
+
+  def self.from_omniauth(auth_info)
+    user = User.find_or_create_by(uid: auth_info[:uid])
```
Comment (javier-aguilar) [inline app/models/user.rb:7]: I believe this is not working as intended. I logged in a few times with my account and had 3 different user accounts

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -1,3 +1,13 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
+  helper_method :current_user
+
+  def current_user
+    @current_user ||= User.find(session[:user_id]) if session[:user_id]
```
Comment (TravisBorgsmiller) [inline app/controllers/application_controller.rb:6]: Good use of memoization

File: app/controllers/sessions_controller.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+  def create
+    user = User.from_omniauth(auth_info)
+    session[:user_id] = user.id
+    redirect_to '/dashboard'
+  end
+  
+  def destroy
+    session.delete(:user_id)
+    # session[:user_id] = nil
```
Comment (TravisBorgsmiller) [inline app/controllers/sessions_controller.rb:13]: We should remove this comment if we don't need it

Comment (SMJ289) [reply app/controllers/sessions_controller.rb]: Got it!

File: app/models/user.rb (hunk 2 above)
Comment (SMJ289) [reply app/models/user.rb]: We added uniqueness validations to the model and that seems to have fixed it. Good catch!

**Comment by SMJ289** [comment]
Rubocop has been configured and the issue with Users should be resolved. @javier-aguilar and @TravisBorgsmiller can you take a quick look, comment, and merge? Thanks!

**Pull request merged**